lib/srv/db/postgres/engine.go

```diff
@@ -33,28 +34,27 @@ import (
 	"github.com/jackc/pgproto3/v2"
 
 	"github.com/gravitational/trace"
-	"github.com/jonboulle/clockwork"
 	"github.com/sirupsen/logrus"
 )
 
+func init() {
+	engineFn := func(ec common.EngineConfig) (common.Engine, error) {
+		return &Engine{
+			EngineConfig: ec,
+		}, nil
+	}
+	common.RegisterEngine(defaults.ProtocolPostgres, engineFn)
+	common.RegisterEngine(defaults.ProtocolCockroachDB, engineFn)
```

Comment:
You could change the order of arguments to use one engine with multiple protocols in `RegisterEngine()`
```go
func RegisterEngine(fn EngineFn, protocols... string)
```
but I'm not sure if this is common enough to worry about it.

Reply:
Good idea, updated.